js_modules/dagit/packages/core/src/workers/fuseSearch.worker.ts

```diff
@@ -0,0 +1,36 @@
+import Fuse from 'fuse.js';
+
+import {SearchResult} from '../search/types';
+
+/**
+ * A Web Worker that creates and queries a Fuse object.
+ */
+
+let fuseObject: null | Fuse<SearchResult[]> = null;
+
+self.addEventListener('message', (event) => {
+  const {data} = event;
+
+  switch (data.type) {
+    case 'set-results': {
+      if (!fuseObject) {
+        fuseObject = new Fuse(data.results, data.fuseOptions);
+      } else {
+        fuseObject.setCollection(data.results);
+      }
+      self.postMessage({type: 'ready'});
+      break;
+    }
+    case 'query': {
+      if (fuseObject) {
+        const {queryString} = data;
+
+        // Consider the empty string as returning no results.
+        const results = queryString ? fuseObject.search(queryString) : [];
+        self.postMessage({type: 'results', queryString, results});
+      }
+    }
+  }
+});
+
+export {};
```

Comment:
This looks good! It's sorta surprising that Fuse is so slow but I guess it does a LOT more than basic substring matching 😅